pkgs/top-level/default.nix

```diff
@@ -60,11 +60,24 @@ in let
 
   localSystem = lib.systems.elaborate args.localSystem;
 
-  # Condition preserves sharing which in turn affects equality.
+  # Condition preserves sharing which in turn improves equality within this Nixpkgs.
+  #
+  # Use `lib.systems.equals` to compare systems.
   crossSystem =
-    if crossSystem0 == null || crossSystem0 == args.localSystem
-    then localSystem
-    else lib.systems.elaborate crossSystem0;
+    if crossSystem0 == null || lib.systems.equals crossSystem0 args.localSystem
```

Comment:
https://gist.github.com/GrahamcOfBorg/1e800ef80702f76206b24b79c9f0dc1d

`error: value is a string while a set was expected`

crossSystem0 and args.localSystem haven't been elaborated so they're just strings

Reply:
> crossSystem0 and args.localSystem haven't been elaborated so they're possibly just strings

IMHO we should elaborate unconditionally here.